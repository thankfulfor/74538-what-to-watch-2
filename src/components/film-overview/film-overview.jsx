import * as React from 'react';
import PropTypes from 'prop-types';

export const FilmOverview = (props) => {
  const {film} = props;
  const {description, rating, director, starring, scores_count: scoresCount} = film;

  const renderTextRating = (score) => {
    switch (true) {
      case (score <= 3):
        return `Bad`;
      case (score <= 5):
        return `Normal`;
      case (score <= 8):
        return `Good`;
      case (score <= 10):
        return `Very good`;
      case (score > 10):
        return `Awesome`;
      default:
        return `No scores yet`;
    }
  };

  return (
    <React.Fragment>
      <div className="movie-rating">
        <div className="movie-rating__score">{rating}</div>
        <p className="movie-rating__meta">
          <span className="movie-rating__level">{renderTextRating(rating)}</span>
          <span className="movie-rating__count">{scoresCount} ratings</span>
        </p>
      </div>

      <div className="movie-card__text">
        <p>{description}</p>

        <p className="movie-card__director"><strong>Director: {director}</strong></p>

        <p className="movie-card__starring">
          <strong>
            Starring:
            {starring.map((star, i) => <React.Fragment key={`${star}-${i}`}>{star}{!!i && `,`}</React.Fragment>)} and other
          </strong>
        </p>
      </div>
    </React.Fragment>
  );
};

FilmOverview.propTypes = {
  film: PropTypes.object.isRequired,
};
