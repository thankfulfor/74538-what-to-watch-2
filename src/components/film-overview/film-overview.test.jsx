import React from 'react';
import renderer from 'react-test-renderer';
import {FilmOverview} from './film-overview.jsx';
import {BrowserRouter} from 'react-router-dom';

const film = {
  "description": `A frontiersman on a fur trading expedition in the 1820s fights for survival after being mauled by a bear and left for dead by members of his own hunting team.`,
  "rating": 4,
  "director": `Alejandro G. Iñárritu`,
  "starring": [
    `Leonardo DiCaprio`,
    `Tom Hardy`,
    `Will Poulter`
  ],
  "scores_count": 618498,
};

it(`FilmOverview корректно рендерится после перезапуска`, () => {
  const tree = renderer
    .create(
        <BrowserRouter>
          <FilmOverview film={film} />
        </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
