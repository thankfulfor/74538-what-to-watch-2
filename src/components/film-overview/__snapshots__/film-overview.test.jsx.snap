// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmOverview корректно рендерится после перезапуска 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      4
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Normal
      </span>
      <span
        className="movie-rating__count"
      >
        618498
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      A frontiersman on a fur trading expedition in the 1820s fights for survival after being mauled by a bear and left for dead by members of his own hunting team.
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Alejandro G. Iñárritu
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring:
        Leonardo DiCaprio
        Tom Hardy
        ,
        Will Poulter
        ,
         and other
      </strong>
    </p>
  </div>,
]
`;
